lib/config/migrations/custom/semantic-prefix-migration.ts

```diff
@@ -0,0 +1,16 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class SemanticPrefixMigration extends AbstractMigration {
+  override readonly deprecated = true;
+  override readonly propertyName = 'semanticPrefix';
+
+  override run(value: unknown): void {
+    if (is.string(value)) {
+      const [text] = value.split(':');
```

Comment:
I think it is better to do this way:
```ts
const semanticCommitMessage = SemanticCommitMessage.fromString(value);
if (SemanticCommitMessage.is(semanticCommitMessage)) {
  const { scope, type } = semanticCommitMessage.toJSON();
  this.setSafely('semanticCommitType', type);
  this.setSafely('semanticCommitScope', is.emptyString(scope) ? null : scope);
} else {
  this.setSafely('semanticCommitType', value);
  this.setSafely('semanticCommitScope', null);
}
```

Reply:
> > > Refactoring code logic using the `SemanticCommitMessage` methods.
> > 
> > 
> > Idea dropped needs discussion. The `SemanticCommitMessage` returns full prefix as type and returns `null` for scope ie. unreliable for now. Eg. for `fix(deps):*` we get `type: fix(deps) , scope:null` instead of `type: fix , scope:deps`
> 
> `fix(deps):*` is not semantic commit message, because it is not follow to specification properly, therefore it will not be parsed by `SemanticCommitMessage.fromString` https://www.conventionalcommits.org/en/v1.0.0/#specification 5. A description MUST immediately follow the colon and **space** after the type/scope prefix.
> 
> I still think would be better to use static method `SemanticCommitMessage.fromString`.

Thanks for clarifying ♥

Reply:
Thanks for clarifying.♥ I didn't know the conventions well.